v3/src/models/history/undo-store.ts

```diff
@@ -7,14 +7,20 @@ import { applyCustomRedo, applyCustomUndo, hasCustomUndoRedo } from "./custom-un
 import { DEBUG_UNDO } from "../../lib/debug"
 
 export interface IUndoManager {
-  undoLevels : number;
-  redoLevels : number;
-  canUndo : boolean;
-  canRedo : boolean;
-  undoEntry : HistoryEntryType | undefined;
-  redoEntry : HistoryEntryType | undefined;
-  undo() : void;
-  redo() : void;
+  undoLevels : number
+  redoLevels : number
+  canUndo : boolean
+  canRedo : boolean
+  undoEntry : HistoryEntryType | undefined
+  redoEntry : HistoryEntryType | undefined
+  undo() : IUndoInformation
+  redo() : IUndoInformation
+}
+
+// Information that is returned by undo and redo calls
+export interface IUndoInformation {
+  id: string|undefined,
+  action: string|undefined
```

Comment:
I would just make them optional:
```suggestion
  id?: string,
  action?: string
```